container/container.go

```diff
@@ -1,201 +1,520 @@
+func (x Container) BasicACL() (res acl.Basic) {
+	res.FromBits(x.v2.GetBasicACL())
+	return
 }
 
-func (c *Container) Version() *version.Version {
-	var ver version.Version
-	if v2ver := c.v2.GetVersion(); v2ver != nil {
-		ver.ReadFromV2(*c.v2.GetVersion())
+// SetPlacementPolicy sets placement policy for the objects within the Container.
+// NeoFS storage layer strives to follow the specified policy.
+//
+// See also PlacementPolicy.
+func (x *Container) SetPlacementPolicy(policy netmap.PlacementPolicy) {
+	var m v2netmap.PlacementPolicy
+	policy.WriteToV2(&m)
+
+	x.v2.SetPlacementPolicy(&m)
+}
+
+// PlacementPolicy returns placement policy set using SetPlacementPolicy.
+//
+// Zero Container has no placement policy which is incorrect according to
+// NeoFS API protocol.
+func (x Container) PlacementPolicy() (res netmap.PlacementPolicy) {
+	m := x.v2.GetPlacementPolicy()
+	if m != nil {
+		err := res.ReadFromV2(*m)
+		if err != nil {
+			panic(fmt.Sprintf("unexpected error from PlacementPolicy.ReadFromV2: %v", err))
+		}
 	}
-	return &ver
+
+	return
 }
 
-func (c *Container) SetVersion(v *version.Version) {
-	var verV2 refs.Version
-	v.WriteToV2(&verV2)
-	c.v2.SetVersion(&verV2)
+// SetAttribute sets Container attribute value by key. Both key and value
+// MUST NOT be empty. Attributes set by the creator (owner) are most commonly
+// ignored by the NeoFS system and used for application layer. Some attributes
+// are so-called system or well-known attributes: they are reserved for system
+// needs. System attributes SHOULD NOT be modified using SetAttribute, use
+// corresponding methods/functions. List of the reserved keys is documented
+// in the particular protocol version.
+//
+// SetAttribute overwrites existing attribute value.
+//
+// See also Attribute, IterateAttributes.
+func (x *Container) SetAttribute(key, value string) {
+	if key == "" {
+		panic("empty attribute key")
+	} else if value == "" {
+		panic("empty attribute value")
+	}
+
+	attrs := x.v2.GetAttributes()
+	ln := len(attrs)
+
+	for i := 0; i < ln; i++ {
+		if attrs[i].GetKey() == key {
+			attrs[i].SetValue(value)
+			return
+		}
+	}
+
+	attrs = append(attrs, container.Attribute{})
+	attrs[ln].SetKey(key)
+	attrs[ln].SetValue(value)
+
+	x.v2.SetAttributes(attrs)
 }
 
-func (c *Container) OwnerID() *user.ID {
-	m := c.v2.GetOwnerID()
-	if m == nil {
-		return nil
+// Attribute reads value of the Container attribute by key. Empty result means
+// attribute absence.
+//
+// See also SetAttribute, IterateAttributes.
+func (x Container) Attribute(key string) string {
+	attrs := x.v2.GetAttributes()
+	for i := range attrs {
+		if attrs[i].GetKey() == key {
+			return attrs[i].GetValue()
+		}
 	}
 
-	var id user.ID
+	return ""
+}
 
-	_ = id.ReadFromV2(*m)
+// IterateAttributes iterates over all Container attributes and passes them
+// into f. The handler MUST NOT be nil.
+//
+// See also SetAttribute, Attribute.
+func (x Container) IterateAttributes(f func(key, val string)) {
```

Comment:
is it possible to remove an attribute somehow?

Reply:
Nope, there is no need now and I can't imagine it.